src/core/opamSystem.ml

```diff
@@ -389,8 +389,25 @@ let dirs dir =
   directories_with_links dir
 
 let dir_is_empty dir =
-  try in_dir dir (fun () -> Sys.readdir (Sys.getcwd ()) = [||])
-  with File_not_found _ -> false
+  try
+    let dir = Unix.opendir dir in
+    Fun.protect ~finally:(fun () -> Unix.closedir dir) @@ fun () ->
+    let rec loop () =
+      match Unix.readdir dir with
```

Comment:
Is there a reason to use `Unix.readdir` isntead of the previous version with Sys.readdir ? is it to retrieve all errors ?

Reply:
Yes, it's much faster. The previous version used 2 system calls + a combination of several system calls + the allocation of an array + a loop to set the elements of the array.

The `Unix.readdir` version simply do 3 system calls: 1 opendir, 1 readdir, 1 closedir

Reply:
ok! :+1: